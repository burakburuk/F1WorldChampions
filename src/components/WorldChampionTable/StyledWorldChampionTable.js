import styled from 'styled-components';

const StyledWorldChampionTable = styled.div`
    width: 100%;
    padding-top: 10px;
    padding-left: 30px;
    padding-right: 30px;

    td {
        border-bottom: 1px solid #9e9e9e;
        height: 50px;
    }
`;

export default StyledWorldChampionTable;
