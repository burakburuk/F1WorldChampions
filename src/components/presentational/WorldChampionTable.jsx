import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
    Table,
    TableHead,
    TableRow,
    TableBody,
    TableCell,
} from './Table';

const StyledWorldChampionTable = styled.div`
    width: 100%;
    padding-top: 10px;
    padding-left: 30px;
    padding-right: 30px;

    td {
        border-bottom: 1px solid #9e9e9e;
        height: 50px;
    }
`;

const WorldChampionTable = props => (
    <StyledWorldChampionTable>
        <Table>
            <TableHead>
                <TableRow>
                    <TableCell>Year</TableCell>
                    <TableCell>Driver</TableCell>
                    <TableCell>Nationality</TableCell>
                    <TableCell>Team</TableCell>
                    <TableCell>Points</TableCell>
                    <TableCell></TableCell>
                </TableRow>
            </TableHead>
            <TableBody>
                {props.champions}
            </TableBody>
        </Table>
    </StyledWorldChampionTable>
);

WorldChampionTable.propTypes = {
    champions: PropTypes.array,
};

export default WorldChampionTable;
