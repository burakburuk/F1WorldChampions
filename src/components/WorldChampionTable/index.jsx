import React from 'react';
import PropTypes from 'prop-types';

import {
    Table,
    TableHead,
    TableRow,
    TableBody,
    TableCell,
} from '../Table';
import StyledWorldChampionTable from './StyledWorldChampionTable';


const WorldChampionTable = props => (
    <StyledWorldChampionTable>
        <Table>
            <TableHead>
                <TableRow>
                    <TableCell>Year</TableCell>
                    <TableCell>Driver</TableCell>
                    <TableCell>Nationality</TableCell>
                    <TableCell>Team</TableCell>
                    <TableCell>Points</TableCell>
                    <TableCell></TableCell>
                </TableRow>
            </TableHead>
            <TableBody>
                {props.champions}
            </TableBody>
        </Table>
    </StyledWorldChampionTable>
);

WorldChampionTable.propTypes = {
    champions: PropTypes.array,
};

export default WorldChampionTable;
