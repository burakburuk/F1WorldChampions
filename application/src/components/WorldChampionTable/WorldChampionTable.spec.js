import React from 'react';
import { shallow } from 'enzyme';
import WorldChampionTable from './';

it('should match with snapshot', () => {
    const wrapper = shallow(<WorldChampionTable />);
    expect(wrapper).toMatchSnapshot();
});
