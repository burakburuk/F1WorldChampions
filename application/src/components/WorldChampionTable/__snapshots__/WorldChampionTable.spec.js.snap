// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<styled.div
  data-id="world-champion-table"
>
  <Table>
    <TableHead>
      <TableRow>
        <TableCell>
          Year
        </TableCell>
        <TableCell>
          Driver
        </TableCell>
        <TableCell>
          Nationality
        </TableCell>
        <TableCell>
          Team
        </TableCell>
        <TableCell>
          Points
        </TableCell>
        <TableCell />
      </TableRow>
    </TableHead>
    <TableBody />
  </Table>
</styled.div>
`;
